### PR TITLE
Secure lead routes

### PR DESCRIPTION
**Pull request by SamardakOleh**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import jwt_decode from "jwt-decode";
 
 import store from "./store.js";
 import setAuthToken from "./utils/setAuthToken.js";
-import setAuthInterceptor from "./utils/setAuthInterceptor.js"
+import setInterceptors from "./utils/setAuthInterceptor"
```
Comment (valentyntu) [inline client/src/App.js]: wrong import

File: client/src/App.js [hunk 2]
```diff
@@ -61,9 +62,9 @@ class App extends Component {
                 path="/lead/:leadId"
                 component={EditLead}
               />
-            </Switch>
-            <Route exact path="/" render={() => this.isUserAuthenticated() ? this.redirectHome(): <Login/>}/>
-            <Route exact path="/register" render={() => this.isUserAuthenticated() ? this.redirectHome(): <Registration/>}/>
+            </Switch>     
+            <Route exact path="/" render={() => this.isUserAuthenticated() ? this.redirectHome(): <Login /> }/>
```
Comment (valentyntu) [inline client/src/App.js]: missing white space before `<Route />` closing

File: client/src/modules/common/NotFound/NotFound.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react'
+import styles from "./NotFound.css";
+import { Navbar } from '../../layouts/Navbar/Navbar';
+export default class NotFound extends Component {
```
Comment (valentyntu) [inline client/src/modules/common/NotFound/NotFound.js]: missing new line between imports and code

File: client/src/modules/lead/editLeadReducer.js [hunk 4]
```diff
@@ -1,6 +1,15 @@
 import leadReducer from "./leadReducer";
 import { combineReducers } from "redux";
+import { NOT_FOUND } from "./types";
+
+const notFoundReducer = (state = false, action) => {
+  if (action.type === NOT_FOUND){ 
+    return action.payload
```
Comment (alexanderko) [inline client/src/modules/lead/editLeadReducer.js]: handle LEAD_LEAD action to reset not found state

File: client/src/modules/lead/leadActions.js [hunk 5]
```diff
@@ -87,13 +87,18 @@ export const loadLead = leadId => dispatch => {
...
+      dispatch(notFoundAction(false));
       dispatch(loadStages(lead.stage.funnel));
     })
     .catch(error => {
-      dispatch({
-        type: GET_ERRORS,
-        payload: error.response.data.errors,
-      });
+      if (error.response.status === 404) {
+        dispatch(notFoundAction(true));
```
Comment (alexanderko) [inline client/src/modules/lead/leadActions.js]: rename `leadNotFound` and remove boolean argument

File: client/src/modules/lead/types.js [hunk 6]
```diff
@@ -6,5 +6,4 @@ export const UPDATE_CONTACT = "UPDATE_CONTACT";
 export const LOAD_LEAD = "LOAD_LEAD";
 export const UPDATE_LEAD = "UPDATE_LEAD";
 export const LOAD_ORGANIZATIONS = "LOAD_ORGANIZATIONS";
-export const CLEAR_LEADS = "CLEAR_LEADS";
-
+export const NOT_FOUND = "NOT_FOUND";
```
Comment (alexanderko) [inline client/src/modules/lead/types.js]: `LEAD_NOT_FOUND`

File: client/src/utils/setAuthInterceptor.js [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
       store.dispatch(logoutUser());
-      window.location.href = "/"
+      history.push("/");
     }
     return Promise.reject(err);
   });
 };
 
-export default setAuthInterceptor;
+export default setAuthInterceptor;
```
Comment (alexanderko) [inline client/src/utils/setAuthInterceptor.js]: new line

File: api/lead/index.js [hunk 8]
```diff
@@ -11,6 +11,28 @@ import Organization from "../../models/organization";
 
 const router = new Router();
 
+const validateLeadDomain = (req, res, next) => {
+  if (process.env.ENV !== "PROD" && req.params.id) {
```
Comment (alexanderko) [inline api/lead/index.js]: extract into separate optional middleware

**Pull request merged**